ZipBuilder/Sources/ZipBuilder/FrameworkBuilder.swift

```diff
@@ -406,32 +419,68 @@ struct FrameworkBuilder {
-      framework != "FirebaseCoreDiagnostics",
-      framework != "FirebaseUI" {
-      let frameworkHeader = headersDir.appendingPathComponent("\(framework).h")
-      guard fileManager.fileExists(atPath: frameworkHeader.path) else {
-        fatalError("Missing explicit umbrella header for \(framework).")
+
+    // Find the location of the public headers.
+    let anyArch = LaunchArgs.shared.archs[0] // any arch is ok, but need to make sure we're building it
+    let archivePath = thinArchives[anyArch]!
+    let headersDir = archivePath.deletingLastPathComponent().appendingPathComponent("Headers")
+
+    // Find CocoaPods generated umbrella header.
+    var umbrellaHeader = ""
+    if framework == "gRPC-Core" {
+      // TODO: Proper handling of podspec-specified module.modulemap files with customized umbrella
+      // headers. This is good enough for Firebase since it doesn't need these modules.
+      umbrellaHeader = "\(framework)-umbrella.h"
+    } else {
+      var umbrellaHeaderURL: URL
+      do {
+        let files = try fileManager.contentsOfDirectory(at: headersDir,
+                                                        includingPropertiesForKeys: nil).compactMap { $0.absoluteString }
+        let umbrellas = files.filter { $0.contains("umbrella.h") }
+        if umbrellas.count != 1 {
+          fatalError("Did not find exactly one umbrella header in \(headersDir).")
+        }
+        umbrellaHeaderURL = URL(string: umbrellas[0])!
+      } catch {
+        fatalError("Error while enumerating files \(headersDir): \(error.localizedDescription)")
+      }
+      // Verify Firebase frameworks include an explicit umbrella header for Firebase.h.
+      if framework.hasPrefix("Firebase"),
+        framework != "FirebaseCoreDiagnostics",
+        framework != "FirebaseUI",
+        !framework.hasSuffix("Swift") {
+        // Delete CocoaPods generated umbrella and use pre-generated one.
+        do {
+          try fileManager.removeItem(at: umbrellaHeaderURL)
+        } catch let error as NSError {
+          print("Failed to delete: \(umbrellaHeaderURL). Error: \(error.domain)")
+        }
+        umbrellaHeader = "\(framework).h"
+        let frameworkHeader = headersDir.appendingPathComponent(umbrellaHeader)
+        guard fileManager.fileExists(atPath: frameworkHeader.path) else {
+          fatalError("Missing explicit umbrella header for \(framework).")
+        }
+      } else {
+        umbrellaHeader = umbrellaHeaderURL.lastPathComponent
       }
     }
-    // Copy the Headers over. Pass in the prefix to remove in order to generate the relative paths
-    // for some frameworks that have nested folders in their public headers.
+    // Copy the Headers over.
     let headersDestination = frameworkDir.appendingPathComponent("Headers")
     do {
-      try recursivelyCopyHeaders(from: headersDir, to: headersDestination)
+      try fileManager.copyItem(at: headersDir, to: headersDestination)
```

Comment:
Why did this change from recurisve?

Reply:
`copyItem` is recursive.  There are no longer any symbolic links in the framework build, so the Swift provided method now meets the requirements.